ToNote/Controls/DragHandler.cs

```diff
@@ -0,0 +1,86 @@
+﻿namespace ToNote.Controls
+{
+    using System.Collections.ObjectModel;
+    using System.Windows;
+    using System.Windows.Controls;
+    using System.Windows.Input;
+    using System.Windows.Media;
+    using ToNote.Models;
+    using ToNote.Views;
+
+    public static class DragHandler
+    {
+        public static void HandleMouseLeftButtonDown(object sender, MouseButtonEventArgs args)
+        {
+            if (args.LeftButton == MouseButtonState.Pressed)
+            {
+                var data = new DataObject();
+                var todoControl = FindAncestorOfType<TodoControl>(sender as DependencyObject);
+                if (todoControl != null)
+                {
+                    data.SetData("Source", todoControl);
+                }
+                else
+                    data.SetData("Source", sender as DependencyObject);
+
+                DragDrop.DoDragDrop(sender as DependencyObject, data, DragDropEffects.Move);
+                args.Handled = true;
+            }
+        }
+
+        public static void HandleDrop(object sender, DragEventArgs args)
+        {
+            var source = args.Data.GetData("Source") as DependencyObject;
+            if (source != null)
+            {
+                if (sender is NoteView noteView)
+                {
+                    var panel = (FindAncestorOfType<ItemsControl>(sender as DependencyObject) as ItemsControl).ItemsSource as ObservableCollection<Note>;
+
+                    var newIndex = panel.IndexOf(noteView.DataContext as Note);
+
+                    var note = (source as NoteView).DataContext as Note;
+
+                    panel.RemoveAt(panel.IndexOf(note));
+
+                    panel.Insert(newIndex, note);
+
+                    args.Handled = true;
+                }
+
+                else
+                {
+                    var panel = FindAncestorOfType<ItemsControl>(sender as DependencyObject) as ItemsControl;
+
+                    var todoControl = FindAncestorOfType<TodoControl>((DependencyObject)sender);
+
+                    int newIndex;
+
+                    if (todoControl != null)
```

Comment:
I'd prefer ternary to avoid uninitialzed variables, ` int newIndex = todoControl != null ? panel.Items.IndexOf(todoControl) : panel.Items.IndexOf(sender);`